lib/utilities/aclUtils.js

```diff
@@ -171,6 +171,10 @@ aclUtils.parseGrant = function parseGrant(grantHeader, grantType) {
     });
 };
 
+aclUtils.isValidCanonicalId = function isValidCanonicalId(canonicalID) {
+    return /^[a-f0-9]{64}$/.test(canonicalID);
```

Comment:
I was wondering about this myself, AWS only generates canonical IDs with a-f in lowercase but I just tested using aws s3api client and it also accepts uppercase letters as a valid id. Thanks, I updated the code.